Source/SuperLinq/CopyTo.cs

```diff
@@ -71,19 +78,30 @@ public static void CopyTo<TSource>(this IEnumerable<TSource> source, TSource[] a
 		Guard.IsNotNull(source);
 		Guard.IsNotNull(array);
 
-		if (TryGetCollectionCount(source, out var n)
-			&& n > array.Length)
+		if (source is ICollection<TSource> coll)
 		{
-			ThrowHelper.ThrowArgumentException(nameof(array), "Destination is not long enough.");
+			coll.CopyTo(array);
 		}
-
-		var i = 0;
-		foreach (var el in source)
+		else if (source is TSource[] arr)
```

Comment:
`TSource[]` implements `ICollection<TSource>`. Remove this or (if it exists for performance) rearrange parameter order.